cylc/flow/network/__init__.py

```diff
@@ -293,22 +273,19 @@ def _socket_options(self):
         self.socket.sndhwm = 10000
 
     def _bespoke_start(self):
-        """Initiate bespoke items on thread at start."""
+        """Initiate bespoke items at start."""
         self.stopping = False
-        sleep(0)  # yield control to other threads
 
     def stop(self, stop_loop=True):
         """Stop the server.
 
         Args:
-            stop_loop (Boolean): Stop running IOLoop of current thread.
+            stop_loop (Boolean): Stop running IOLoop.
 
         """
         self._bespoke_stop()
         if stop_loop and self.loop and self.loop.is_running():
             self.loop.stop()
```

Comment:
In my debugging, I have found that `self.loop` is still running after the call to `self.loop.stop()`. Any idea why this is the case? (Not specific to this branch, also happens on master).

Reply:
Probably waiting for loop to be clear of tasks.. What's still running is probably the question..
Can wait for them to complete after the call to stop:
```
    # Let's also finish all running tasks:
    pending = asyncio.Task.all_tasks()
    loop.run_until_complete(asyncio.gather(*pending))
```
or cancel them
```
    # Let's also cancel all running tasks:
    pending = asyncio.Task.all_tasks()
    for task in pending:
        task.cancel()
        # Now we should await task to execute it's cancellation.
        # Cancelled task raises asyncio.CancelledError that we can suppress:
        with suppress(asyncio.CancelledError):
            loop.run_until_complete(task)
```
(from these:
https://stackoverflow.com/a/37430948
https://stackoverflow.com/a/37345564
)

Reply:
Will give it a try

Reply:
how did you track the running processes on shutdown by the way? pycharm?

Reply:
I tried `asyncio.all_tasks(self.loop)` and yes there was a task `coro=<WorkflowSubscriber.subscribe() ...` still running. The scenario was: one workflow played then stopped in the UI. I didn't track anything all the way to shutdown, just inspected the loop while stopping the workflow (in VSCode). But by keeping a reference to the loop I noticed that after stopping it, playing and stopping a different workflow, the loop for the first workflow subscription was still running.